### PR TITLE
Add readonly typings for Set and Map

### LINKED ISSUE #10919: Add readonly typings for Set and Map similar to ReadonlyArray
Array has a companion [ReadonlyArray](https://github.com/Microsoft/TypeScript/blob/fe642f558633d8ea0cbfa75cc7e0a82c6d7ae2aa/src/lib/es2015.core.d.ts#L350) type with all the mutation methods removed. There should be similar versions for the es6 collections Set and Map in [es2015.collection.d.ts](https://github.com/Microsoft/TypeScript/blob/fe642f558633d8ea0cbfa75cc7e0a82c6d7ae2aa/src/lib/es2015.collection.d.ts).

The benefit of this change is that it would allow you to express a collection which is not changed after initialization

### PR DESCRIPTION
**Pull request by gmoothart**
Fixes #10919

Similar to ReadonlyArray, these typings remove the mutation methods from Set and Map so that they can only be initialized by the constructor.

[Example](https://www.typescriptlang.org/play/#src=interface%20ReadonlyMap%3CK%2C%20V%3E%20%7B%0D%0A%20%20forEach(callbackfn%3A%20(value%3A%20V%2C%20index%3A%20K%2C%20map%3A%20ReadonlyMap%3CK%2C%20V%3E)%20%3D%3E%20void%2C%20thisArg%3F%3A%20any)%3A%20void%3B%0D%0A%20%20get(key%3A%20K)%3A%20V%7Cundefined%3B%0D%0A%20%20has(key%3A%20K)%3A%20boolean%3B%0D%0A%20%20readonly%20size%3A%20number%3B%0D%0A%7D%0D%0A%0D%0Ainterface%20ReadonlySet%3CT%3E%20%7B%0D%0A%20%20forEach(callbackfn%3A%20(value%3A%20T%2C%20index%3A%20T%2C%20set%3A%20ReadonlySet%3CT%3E)%20%3D%3E%20void%2C%20thisArg%3F%3A%20any)%3A%0D%0A%20%20%20%20%20%20void%3B%0D%0A%20%20has(value%3A%20T)%3A%20boolean%3B%0D%0A%20%20readonly%20size%3A%20number%3B%0D%0A%7D%0D%0A%0D%0Alet%20roMap%3A%20ReadonlyMap%3Cstring%2C%20string%3E%20%3D%20new%20Map(%0D%0A%09%5B%5B%22key1%22%2C%20%22value1%22%5D%2C%20%5B%22key2%22%2C%20%22value2%22%5D%5D)%3B%0D%0Alet%20roSet%3A%20ReadonlySet%3Cnumber%3E%20%3D%20new%20Set(%5B1%2C2%2C3%2C4%2C4%5D)%3B%0D%0A%0D%0A%0D%0Alet%20v1%20%3D%20roMap.get(%22key1%22)%3B%0D%0A%2F%2F%20not%20available%3A%0D%0A%2F%2FroMap.set(%22key1%22%2C%20%22foo%22)%3B%0D%0A%0D%0Alet%20b%20%3D%20roSet.has(5)%3B%0D%0A%2F%2F%20not%20available%3A%0D%0A%2F%2FroSet.delete(4)%3B%0D%0A)

### CODE REVIEW

**Comment by mhegazy** [comment]
thanks @gmoothart. can you please sign the CLA?

File: src/lib/es2015.collection.d.ts [hunk 1]
```diff
@@ -15,6 +15,15 @@ interface MapConstructor {
 }
 declare var Map: MapConstructor;
 
+interface ReadonlyMap<K, V> {
+  forEach(
+      callbackfn: (value: V, key: K, map: ReadonlyMap<K, V>) => void,
+      thisArg?: any): void;
+  get(key: K): V|undefined;
+  has(key: K): boolean;
+  readonly size: number;
```
Comment (mhegazy) [inline src/lib/es2015.collection.d.ts]: Please match the formatting for the rest of the file, that is 4 spaces.

File: src/lib/es2015.collection.d.ts [hunk 2]
```diff
@@ -15,6 +15,15 @@ interface MapConstructor {
 }
 declare var Map: MapConstructor;
 
+interface ReadonlyMap<K, V> {
+  forEach(
+      callbackfn: (value: V, key: K, map: ReadonlyMap<K, V>) => void,
+      thisArg?: any): void;
+  get(key: K): V|undefined;
```
Comment (mhegazy) [inline src/lib/es2015.collection.d.ts]: and the spacing as well.

**Comment by mhegazy** [comment]
I guess for the CLA i need to talk to @naomiblack.

**Comment by gmoothart** [comment]
Formatting fixed. I also updated some parameters names which were wrong, and fixed the example.

It would be nice to be able to have type inference for the generic parameters, but I don't think it's possible in this case?

Re: the CLA, there is a custom Google/Microsoft one that I did sign and yeah, Naomi should be able to help you.

**Pull request merged**